integration_tests/src/utils.rs

```diff
@@ -261,3 +324,370 @@ pub fn set_sigterm() {
+    };
+
+    let proposal = RegisterCoinProposal {
+        title: coin_params.proposal_title,
+        description: coin_params.proposal_desc,
+        metadata: Some(coin_params.coin_metadata),
+    };
+    let res = contact
+        .submit_register_coin_proposal(
+            proposal,
+            get_deposit(),
+            get_fee(None),
+            keys[0].validator_key,
+            Some(duration),
+        )
+        .await
+        .unwrap();
+    info!("Gov proposal executed with {:?}", res.raw_log);
+
+    vote_yes_on_proposals(contact, keys, None).await;
+    wait_for_proposals_to_execute(contact).await;
+}
+
+fn parse_phrases(filename: &str) -> (Vec<CosmosPrivateKey>, Vec<String>) {
+    let file = File::open(filename).expect("Failed to find phrases");
+    let reader = BufReader::new(file);
+    let mut ret_keys = Vec::new();
+    let mut ret_phrases = Vec::new();
+
+    for line in reader.lines() {
+        let phrase = line.expect("Error reading phrase file!");
+        if phrase.is_empty()
+            || phrase.contains("write this mnemonic phrase")
+            || phrase.contains("recover your account if")
+        {
+            continue;
+        }
+        let key = CosmosPrivateKey::from_phrase(&phrase, "").expect("Bad phrase!");
+        ret_keys.push(key);
+        ret_phrases.push(phrase);
+    }
+    (ret_keys, ret_phrases)
+}
+
+pub fn parse_validator_keys() -> (Vec<CosmosPrivateKey>, Vec<String>) {
+    let filename = "/validator-phrases";
+    info!("Reading mnemonics from {}", filename);
+    parse_phrases(filename)
+}
+
+pub fn get_keys() -> Vec<ValidatorKeys> {
+    let (cosmos_keys, cosmos_phrases) = parse_validator_keys();
+    let mut ret = Vec::new();
+    for (c_key, c_phrase) in cosmos_keys.into_iter().zip(cosmos_phrases) {
+        ret.push(ValidatorKeys {
+            validator_key: c_key,
+            validator_phrase: c_phrase,
+        })
+    }
+    ret
+}
+
+pub async fn send_althea_tokens(addresses: Vec<AltheaAddress>) {
+    // Create a contact object and get our balance
+    let althea_contact = Contact::new(
+        &get_althea_grpc(),
+        ALTHEA_CONTACT_TIMEOUT,
+        ALTHEA_CHAIN_PREFIX,
+    )
+    .unwrap();
+
+    //register uUSDC as a ERC20
+    let mut bank_qc = QueryClient::connect(althea_contact.get_url())
+        .await
+        .expect("Unable to connect to bank query client");
+    let metadata = bank_qc
+        .denom_metadata(QueryDenomMetadataRequest {
+            denom: "uUSDC".to_string(),
+        })
+        .await
+        .expect("Unable to query denom metadata")
+        .into_inner()
+        .metadata
+        .expect("No metadata for erc20 coin");
+
+    let coin_params = RegisterCoinProposalParams {
+        coin_metadata: metadata.clone(),
+        proposal_desc: "Register Coin Proposal Description".to_string(),
+        proposal_title: "Register Coin Proposal Title".to_string(),
+    };
+    execute_register_coin_proposal(
```

Comment:
This looks like it's always running the coin proposal even if it's already been passed, that's going to really increase the amount of time it takes to run this function from a few seconds to two minutes or so for no particularly good reason.

Reply:
I separated this into its own function